bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/handler/InsteonNetworkHandler.java

```diff
@@ -73,6 +84,18 @@ public void initialize() {
         updateStatus(ThingStatus.UNKNOWN);
 
         scheduler.execute(() -> {
+            synchronized (LOCK) {
```

Comment:
If there is a better way to do this, let me know. I asked for help on the community forum (https://community.openhab.org/t/ensuring-a-single-bridge-per-binding/94928), but didn't get any responses.

Reply:
Why are you trying to enforce a single bridge?
I think the problem in this particular case is that what you really expect is a single instance of `InsteonBinding`. The proper way to enforce that would be through a OSGI service like managing the InsteonBinding lifecycle from within the ThingHandlerFactory.

Reply:
Another option you have is instead of the console extension trying to get an instance of InsteonNetworkHandler you have instances of InsteonNetworkHandler register themselves with the console extension. That should be easy enough by registering InsteonNetworkHandler instances as they are created in ThingHandlerFactory. ThingHandlerFactory can easy get the console extension as a reference.

Reply:
There is only a single network that can exist anyway the way the Insteon protocol works (via electrical wires and wirelessly), and the ported code base assumes only a single network.

Reply:
@cpmeister, If possible I'd like to get this change included in the 2.4 release. It will be helpful as people migrate from the OH1 binding. I also noticed the other reviewers were removed and you are the only one left.

Reply:
This has been removed